### PR TITLE
2.4 PXB-2290 (Merge 5.7.31 upstream)

### PR DESCRIPTION
**Pull request by altmannmarcelo**
https://jira.percona.com/browse/PXB-2290

Adjusted innodb initialization to call os_event_global_init() due to
change introduced at 253d846

### CODE REVIEW

**Comment by altmannmarcelo** [comment]
https://pxb.cd.percona.com/job/percona-xtrabackup-2.4-param-medium/113/

```
t/bug1135431.sh
t/bug1623210.sh
t/xb_databases_options.sh
```

Those are the test that are failing on centos6. I'm waiting for PXB-2261 in order to further troubleshoot it.

```
t/partition.sh
```
This is PXB-2292

```
t/mdl_locks.sh
```
Unstable test after PXB-2254

**Comment by rahulmalik87** [comment]
could you run the test suite with 5.7.30? you may be need to bump the version in boostrap.sh

File: storage/innobase/xtrabackup/src/backup_copy.cc [hunk 1]
```diff
@@ -2082,6 +2082,7 @@ copy_back(int argc, char **argv)
 	srv_page_size_shift = 14;
 	srv_page_size = (1 << srv_page_size_shift);
 	srv_max_n_threads = 1000;
+  os_event_global_init();
```
Comment (satya-bodapati) [inline storage/innobase/xtrabackup/src/backup_copy.cc]: can you check alignment?

File: storage/innobase/xtrabackup/src/xtrabackup.cc [hunk 2]
```diff
@@ -2138,7 +2138,7 @@ static my_bool
 innodb_init(void)
 {
 	int	err;
-
+  os_event_global_init();
```
Comment (satya-bodapati) [inline storage/innobase/xtrabackup/src/xtrabackup.cc]: alignment here and other palces too.. I guess it uses tabs instead of spaces

**Review by satya-bodapati — CHANGES_REQUESTED** [review]

File: storage/innobase/xtrabackup/src/xtrabackup.cc (hunk 2 above)
Comment (altmannmarcelo) [reply storage/innobase/xtrabackup/src/xtrabackup.cc]: @satya-bodapati fixed

**Comment by altmannmarcelo** [comment]
@rahulmalik87 https://pxb.cd.percona.com/job/percona-xtrabackup-2.4-param-medium/117/

File: storage/innobase/xtrabackup/test/run.sh [hunk 3]
```diff
@@ -283,6 +283,8 @@ function set_vars()
     then
         PATH="$PWD/..:$PWD/../src:$PWD/../src/xbcloud:$PATH"
         XTRABACKUP_BASEDIR="$PWD/../src"
+    else
+      XTRABACKUP_BASEDIR="$PWD/../"
```
Comment (satya-bodapati) [inline storage/innobase/xtrabackup/test/run.sh]: What does this allow us to do?

Comment (altmannmarcelo) [reply storage/innobase/xtrabackup/test/run.sh]: `XTRABACKUP_BASEDIR` is used by keyring tests. Jenkins run this on an in-source build, however, if you use this on a folder of result from `make install` SRC folder is not present. This change sets `XTRABACKUP_BASEDIR` on those cases.

Also:
storage/innobase/xtrabackup/test/inc/keyring_file.sh
storage/innobase/xtrabackup/test/inc/keyring_vault.sh
```
if test -d ${XTRABACKUP_BASEDIR}/keyring
then
  plugin_dir=${XTRABACKUP_BASEDIR}/keyring
else
  plugin_dir=${XTRABACKUP_BASEDIR}/lib/plugin
fi
```
Added the same validation on test cases to use either keyring folder(in-source build) or lib/plugin

**Comment by satya-bodapati** [comment]
> https://pxb.cd.percona.com/job/percona-xtrabackup-2.4-param-medium/113/
> 
> ```
> t/bug1135431.sh
> t/bug1623210.sh
> t/xb_databases_options.sh
> ```
> 
> Those are the test that are failing on centos6. I'm waiting for PXB-2261 in order to further troubleshoot it.
> 
> ```
> t/partition.sh
> ```
> 
> This is PXB-2292
> 
> ```
> t/mdl_locks.sh
> ```
> 
> Unstable test after PXB-2254

@altmannmarcelo These tests fail consistently 
t/xb_databases_options.sh
t/pxb-1936.sh
t/bug1135431.sh
t/bug1623210.sh

https://pxb.cd.percona.com/job/percona-xtrabackup-2.4-param-medium/117/#showFailuresLink

Can we clean them up as part of Merge? (Because we don't touch 2.4 frequently)

Ask @rahulmalik87 too, He wants/willing to fix some of them

**Review by satya-bodapati — APPROVED** [review]
lets deal those existing centos issues later

**Review by rahulmalik87 — APPROVED** [review]
LGTM

**Pull request merged**